parser/strconv.go

```diff
@@ -16,6 +17,32 @@ func ParseTime(lit string) (*ast.DateTimeLiteral, error) {
 	}, nil
 }
 
+// ParseSignedDuration will convert a string into a possibly negative DurationLiteral.
+func ParseSignedDuration(lit string) (*ast.DurationLiteral, error) {
+	if len(lit) > 0 && lit[0] == '-' {
+		d, err := parser.ParseDuration(lit[1:])
```

Comment:
I would prefer this to consume the local ParseDuration method instead of the one in the parser package. That way the logic to construct the ast node is not duplicated.